libres/lib/analysis/ies/ies_enkf.cpp

```diff
@@ -93,182 +83,103 @@ static void printf_mask(FILE *log_fp, const char *name,
     fputs("\n", log_fp);
 }
 
-void ies_enkf_init_update(void *arg, const bool_vector_type *ens_mask,
-                          const bool_vector_type *obs_mask,
-                          const matrix_type *S, const matrix_type *R,
-                          const matrix_type *dObs, const matrix_type *E,
-                          const matrix_type *D, rng_type *rng) {
-    ies_enkf_data_type *module_data = ies_enkf_data_safe_cast(arg);
-
-    /* Store current ens_mask in module_data->ens_mask for each iteration */
-    ies_enkf_data_update_ens_mask(module_data, ens_mask);
-
-    /* Store obs_mask for initial iteration in module_data->obs_mask0,
-*  for each subsequent iteration we store the current mask in module_data->obs_mask */
-    ies_enkf_store_initial_obs_mask(module_data, obs_mask);
-    ies_enkf_update_obs_mask(module_data, obs_mask);
-}
-
-void ies_enkf_updateA(
-    void *module_data,
-    matrix_type *A,          // Updated ensemble A retured to ERT.
-    const matrix_type *Yin,  // Ensemble of predicted measurements
-    const matrix_type *Rin,  // Measurement error covariance matrix (not used)
-    const matrix_type *dObs, // Actual observations (not used)
-    const matrix_type *Ein,  // Ensemble of observation perturbations
-    const matrix_type *Din,  // (d+E-Y) Ensemble of perturbed observations - Y
-    rng_type *rng) {
-
-    ies_enkf_data_type *data = ies_enkf_data_safe_cast(module_data);
-    const ies_enkf_config_type *ies_config = ies_enkf_data_get_config(data);
-
-    int nrobs_msk =
-        ies_enkf_data_get_obs_mask_size(data); // Total number of observations
-    int nrobs_inp = matrix_get_rows(
-        Yin); // Number of active observations input in current iteration
-
-    int ens_size_msk =
-        ies_enkf_data_get_ens_mask_size(data); // Total number of realizations
-    int ens_size = matrix_get_columns(
-        Yin); // Number of active realizations in current iteration
-    int state_size = matrix_get_rows(A);
-
-    ies_inversion_type ies_inversion =
-        ies_enkf_config_get_ies_inversion(ies_config);
-    double truncation = ies_enkf_config_get_truncation(ies_config);
-    bool ies_debug = ies_enkf_config_get_ies_debug(ies_config);
-    int subspace_dimension =
-        ies_enkf_config_get_enkf_subspace_dimension(ies_config);
-    double rcond = 0;
-    int nrsing = 0;
-    int iteration_nr = ies_enkf_data_inc_iteration_nr(data);
-    FILE *log_fp;
+static void log_active(FILE *log_fp, int ens_size, int ens_size_msk,
+                       int nrobs_input, int nrobs_msk, int state_size,
+                       const ies_enkf_data_type &data) {
+    fprintf(log_fp, "----active ens_size  = %d, total ens_size_msk = %d\n",
```

Comment:
We have a logging utility now which should channel logs to the correct place. Is there a way of replacing these `fprintf`s with that now @dotfloat.

Reply:
```cpp
#include <ert/python.hpp>

namespace {
auto logger = ert::get_logger("ies");
} // namespace

[...]

logger->info("----active ens_size = {}, total end_size_msk = {}", ens_size, ens_size_msk);
```